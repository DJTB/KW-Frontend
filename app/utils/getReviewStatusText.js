import { format, isPast, isFuture, distanceInWordsToNow } from 'date-fns';

import { DATE_FORMAT } from 'shared/constants';

function getReviewStatusText({ vacationDate, reviewsCount, nextReviewDate }) {
  if (vacationDate) {
    return `On Vacation since ${format(vacationDate, DATE_FORMAT)}`;
  }
  if (reviewsCount < 1 && nextReviewDate === false) {
    return 'Next Review: No reviews unlocked';
  }
  if (isPast(nextReviewDate)) {
    return 'Next Review: Now!';
  }
  if (isFuture(nextReviewDate)) {
    return `Next Review: ${distanceInWordsToNow(nextReviewDate, { includeSeconds: true, suffix: true })}`;
  }
  return 'Next Review: Unknown';
}

export default getReviewStatusText;
