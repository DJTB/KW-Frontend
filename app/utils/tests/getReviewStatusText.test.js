import addHours from 'date-fns/add_hours';
import subHours from 'date-fns/sub_hours';
import getReviewStatusText from '../getReviewStatusText';

describe('getReviewStatusText()', () => {
  it('should have sane default', () => {
    expect(getReviewStatusText({})).toMatchSnapshot();
  });
  it('on vacation', () => {
    expect(getReviewStatusText({
      vacationDate: new Date(2017, 11, 26), // Dec 26 2017
      reviewsCount: 40,
      nextReviewDate: false,
    })).toMatchSnapshot();
  });
  it('no reviews unlocked', () => {
    expect(getReviewStatusText({ reviewsCount: 0, nextReviewDate: false })).toMatchSnapshot();
  });
  it('review date is past', () => {
    expect(getReviewStatusText({ reviewsCount: 0, nextReviewDate: subHours(new Date(), 2) })).toMatchSnapshot();
  });
  it('review date is future', () => {
    expect(getReviewStatusText({ reviewsCount: 0, nextReviewDate: addHours(new Date(), 2) })).toMatchSnapshot();
  });
});
