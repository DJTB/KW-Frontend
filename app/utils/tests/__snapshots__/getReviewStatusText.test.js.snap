// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getReviewStatusText() no reviews unlocked 1`] = `"Next Review: No reviews unlocked"`;

exports[`getReviewStatusText() on vacation 1`] = `"On Vacation since Dec 26th 2017"`;

exports[`getReviewStatusText() review date is future 1`] = `"Next Review: about 2 hours"`;

exports[`getReviewStatusText() review date is past 1`] = `"Next Review: Now!"`;

exports[`getReviewStatusText() should have sane default 1`] = `"Next Review: Unknown"`;
